### PR TITLE
Unify masterbars in Simple and Atomic Default sites (take 3)

### PR DESCRIPTION
**Pull request by fushar**
Related to:

- https://github.com/Automattic/dotcom-forge/issues/8314
- https://github.com/Automattic/jetpack/pull/38377#issuecomment-2238010469
- https://github.com/Automattic/jetpack/pull/38399
- https://github.com/Automattic/jetpack/pull/38362

## Proposed changes:

- For Simple sites: this fixes hamburger menu on mobile view on Simple Default on wp-admin pages.
   - It was not working because we added an unnecessary JS file.
- For Atomic sites: this actually shows the Core masterbar on Atomic Default wp-admin pages.
   - It was not working because we reordered the `wpcom-admin-bar` feature in https://github.com/Automattic/jetpack/pull/38364, which contains the `WPCOM_ADMIN_BAR_UNIFICATION`.
   - (in other words, it's a "logical merge conflict")
 - For both Simple and Atomic sites:
   - Fix the W logo not doing anything on wp-admin pages.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Test all (yes, ALL) possible combinations of the following (there are 2 x 2 x 3 x 2):

- Simple / Atomic
- Default / Classic
- Calypso page / wp-admin page / site frontend page
- Desktop / mobile views

In each case:
- You should see Core / "Core-like" admin bar.

When testing mobile view on wp-admin/site frontend:
- REFRESH the page first before testing (to make sure your browser actually treat it as mobile)
- Test whether the hamburger icon can toggle the sidebar (in /wp-admin)
- Test whether the W logo can be clickable to /sites

When testing Calypso page:
- Test whether the sidebar menu is still correct (conforming to Default or Classic)

### CODE REVIEW

**Comment by okmttdhr** [comment]
- Atomic
	- Classic
		- Desktop 
			- wp-admin page: seems ✅ 
				Screenshot 2024-07-19 at 13 01 11
			- /overview page: seems ✅ 
				Screenshot 2024-07-19 at 12 59 03
			- frontend: seems ✅ 
				Screenshot 2024-07-19 at 12 59 45
		- Mobile
			- wp-admin page: Clicking the W icon does not take me to /sites. Is this expected?
				Screenshot 2024-07-19 at 12 56 13
			- /overview page: Do we need 🏠 & ➕ ?
				Screenshot 2024-07-19 at 12 58 15
			- frontend: seems ✅ 
				Screenshot 2024-07-19 at 12 59 59

I'll try testing more...

**Comment by fushar** [comment]
> wp-admin page: Clicking the W icon does not take me to /sites. Is this expected?

No, that's what I've been trying to debug, it's not ready to review actually, sorry for false alarm 😂

**Comment by okmttdhr** [comment]
Re-tested;

- Atomic
	- Classic
		- Desktop ✅ 
		- Mobile
			- wp-admin page: Clicking the W icon took me to /sites ✅ 
				Screenshot 2024-07-19 at 14 04 49
			- /overview page: Do we need 🏠 & ➕ ? 🏠 can take us back to dashboard. 
				Screenshot 2024-07-19 at 14 05 50
			- frontend: seems ✅ 
				Screenshot 2024-07-19 at 14 06 25
	- Default
		- Desktop
			- wp-admin page: the same as Classic ✅ 
			- /overview page: the same as Classic ✅ 
			- frontend: the same as Classic ✅ 
		- Mobile
			- wp-admin page: the same as Classic ✅ 
			- /overview page: Do we need 🏠 & ➕ ?
				Screenshot 2024-07-19 at 14 09 34
			- frontend: the same as Classic ✅

**Comment by fushar** [comment]
> /overview page: Do we need 🏠 & ➕ ?

As this is a Calypso page, we can follow-up with it later :)

**Comment by okmttdhr** [comment]
- Simple
	- Classic
		- Desktop 
			- wp-admin page ✅
				Screenshot 2024-07-19 at 14 49 40
			- /overview page ✅ 
				Screenshot 2024-07-19 at 14 50 12
			- frontend ✅
				Screenshot 2024-07-19 at 14 50 51
		- Mobile
			- wp-admin page ✅
				Screenshot 2024-07-19 at 14 52 05
			- /overview page: the same as Atomic
				Screenshot 2024-07-19 at 14 52 38
			- frontend ✅
				Screenshot 2024-07-19 at 14 51 30

**Comment by okmttdhr** [comment]
- Simple
	- Default
		- Desktop
			- wp-admin page: the same as Simple Classic✅
			- /overview page: the same as Simple Classic✅
			- frontend: the same as Simple Classic✅
		- Mobile
			- wp-admin page: the same as Simple Classic✅
			- /overview page: the same as Atomic
			- frontend: the same as Simple Classic✅

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.scss [hunk 1]
```diff
@@ -17,7 +17,13 @@
 /**
  * WP logo menu
  */
-#wpadminbar #wp-admin-bar-wp-logo>.ab-item .ab-icon:before {
+#wpadminbar #wp-admin-bar-wpcom-logo>.ab-item .ab-icon {
+	width: 15px;
```
Comment (okmttdhr) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.scss:21]: took style from https://github.com/WordPress/wordpress-develop/blob/5a30482419f1b0bcc713a7fdee3a14afd67a1bca/src/wp-includes/css/admin-bar.css#L493 to resolve the broken CSS layout of the wpcom logo

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.scss [hunk 2]
```diff
@@ -27,6 +33,7 @@
...
 	mask-position: center;
 	mask-repeat: no-repeat;
+	top: 2px;
```
Comment (okmttdhr) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.scss:36]: took style from https://github.com/WordPress/wordpress-develop/blob/5a30482419f1b0bcc713a7fdee3a14afd67a1bca/src/wp-includes/css/admin-bar.css#L505 to resolve the broken CSS layout of the wpcom logo

**Comment by fushar** [comment]
Does top: 2px work on mobile view? Just wanted to make sure

**Comment by okmttdhr** [comment]
> Does top: 2px work on mobile view? Just wanted to make sure

It's overridden on mobile and positioning seems to work fine:
Screenshot 2024-07-19 at 15 36 21
Screenshot 2024-07-19 at 15 36 27

**Review by okmttdhr — APPROVED** [review]
Looks good to me now ✅ 

>/overview page: Do we need 🏠 & ➕ ? 🏠 can take us back to dashboard.

As a follow-up, let's display 🏠 & ➕ and hide ❓& 🔔 on mobile.

**Comment by fushar** [comment]
Thank you! Feel free to merge if we're confident.

**Pull request merged**